### PR TITLE
Peer self review history

### PR DESCRIPTION
**Pull request by RakshandaWankhede**
Peer Review History self Review History code - done

### CODE REVIEW

File: client/src/components/PeerReviewDetails/PeerReviewDetails.js [hunk 1]
```diff
@@ -47,18 +47,18 @@ const PeerReviewDetails = props => {
       [
         <span className={classes.boldFont}>Employee Under Review</span>,
         reviewData.employee_under_review.firstname +
-          ' ' +
-          reviewData.employee_under_review.lastname,
+        ' ' +
+        reviewData.employee_under_review.lastname,
```
Comment (shivamsharma30) [inline client/src/components/PeerReviewDetails/PeerReviewDetails.js:51]: I think we need to use template literal here. Is it possible to use template literal?

File: client/src/views/PeerReviewHistory/PeerReviewHistory.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    if (peerReviews) {
+        filteredReview = peerReviews.filter(
+            cls =>
+                cls.status === 'Done'
+        )
+        filteredReview.map((review) => {
+            peerReviewsArray.push([
+                review.employee_under_review.firstname +
+                ' ' +
+                review.employee_under_review.lastname,
```
Comment (shivamsharma30) [inline client/src/views/PeerReviewHistory/PeerReviewHistory.js:62]: I think we need to use template literal here. Is it possible to use template literal?

File: client/src/components/SelfReviewForm/SelfReviewForm.js [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+          submitReview(values)
+          setSubmitting(false)
+        }}
+        validationSchema={Yup.object(validationSchema)}
+      >
+        {({ isSubmitting, values, setFieldValue, handleChange }) => (
+          <Card>
+            <Form>
+              <CardHeader color="primary">
+                <h4 className={classes.cardTitleWhite}>CREATE SELF REVIEW FORM</h4>
```
Comment (shivamsharma30) [inline client/src/components/SelfReviewForm/SelfReviewForm.js]: can we Destructuring className in side the component?

File: client/src/views/PeerReviewHistory/PeerReviewHistory.js (hunk 2 above)
Comment (RakshandaWankhede) [reply client/src/views/PeerReviewHistory/PeerReviewHistory.js:62]: Did in PR #31

File: client/src/components/PeerReviewDetails/PeerReviewDetails.js (hunk 1 above)
Comment (RakshandaWankhede) [reply client/src/components/PeerReviewDetails/PeerReviewDetails.js:51]: Did in PR #31

File: client/src/components/SelfReviewForm/SelfReviewForm.js [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+                  </Grid>
+                  <Grid xs={6} sm={6} md={3} item>
+                    <SelectMenu
+                      name="employee"
+                      onChange={handleChange}
+                      disabledName="Select Employee"
+                      value={values.employee}
+                    >
+                      {employeeData
+                        ? employeeData.map((prop, key) => {
```
Comment (shivamsharma30) [inline client/src/components/SelfReviewForm/SelfReviewForm.js]: if you want render only true condition.  you can use && (AND) operator. also you no need to return **null**.
EX. {employeeData
        && employeeData.map((prop, key) => {
          return prop.status !== 'Inactive' && (
          )
        })
        }

File: client/src/components/SelfReviewForm/SelfReviewForm.js [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+                    <SelectMenu
+                      name="projects"
+                      onChange={handleChange}
+                      disabledName="Select Project"
+                      value={values.projects}
+                      multiple
+                    >
+                      {projectsData
+                        ? projectsData.map((prop, key) => {
+                          return (
```
Comment (shivamsharma30) [inline client/src/components/SelfReviewForm/SelfReviewForm.js:188]: if you want render only true condition. you can use && (AND) operator. also you no need to return null.

File: client/src/components/SelfReviewForm/SelfReviewForm.js [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+                  </Grid>
+                  <Grid xs={6} sm={6} md={3} item>
+                    <SelectMenu
+                      name="functional_manager"
+                      onChange={handleChange}
+                      disabledName="Select Manager"
+                      value={values.functional_manager}
+                    >
+                      {managers
+                        ? managers.map(item => {
```
Comment (shivamsharma30) [inline client/src/components/SelfReviewForm/SelfReviewForm.js]: if you want render only true condition. you can use && (AND) operator. also you no need to return null.

File: client/src/components/SelfReviewForm/SelfReviewForm.js [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+                          return (
+                            <MenuItem
+                              value={item._id}
+                              className={classes.hoverEffect}
+                            >
+                              {item.firstname + ' ' + item.lastname}
```
Comment (shivamsharma30) [inline client/src/components/SelfReviewForm/SelfReviewForm.js]: I think we need to use template literal here. Is it possible to use template literal?

File: client/src/components/selfReviewDetails/SelfReviewDetails.js [hunk 8]
```diff
@@ -147,10 +149,10 @@ const SelfReviewDetails = props => {
           </Button>
 
           {/* className={classes.disabledButton} */}
```
Comment (shivamsharma30) [inline client/src/components/selfReviewDetails/SelfReviewDetails.js]: please remove this commented code.

File: server/app/api/controllers/users.js [hunk 9]
```diff
@@ -113,7 +113,7 @@ module.exports = {
     });
   },
   getAll: function (req, res, next) {
-    const query = req.query? req.query: {}
+    const query = req.query ? req.query : {}
```
Comment (shivamsharma30) [inline server/app/api/controllers/users.js]: Please use || (OR) operator.
Ex.  **const query =  req.query || {}**

File: client/src/components/SelfReviewForm/SelfReviewForm.js (hunk 3 above)
Comment (RakshandaWankhede) [reply client/src/components/SelfReviewForm/SelfReviewForm.js]: Done

File: client/src/components/SelfReviewForm/SelfReviewForm.js (hunk 4 above)
Comment (RakshandaWankhede) [reply client/src/components/SelfReviewForm/SelfReviewForm.js]: Done

File: client/src/components/SelfReviewForm/SelfReviewForm.js (hunk 5 above)
Comment (RakshandaWankhede) [reply client/src/components/SelfReviewForm/SelfReviewForm.js:188]: done

File: client/src/components/SelfReviewForm/SelfReviewForm.js (hunk 6 above)
Comment (RakshandaWankhede) [reply client/src/components/SelfReviewForm/SelfReviewForm.js]: done

File: client/src/components/SelfReviewForm/SelfReviewForm.js (hunk 7 above)
Comment (RakshandaWankhede) [reply client/src/components/SelfReviewForm/SelfReviewForm.js]: done

File: client/src/components/selfReviewDetails/SelfReviewDetails.js (hunk 8 above)
Comment (RakshandaWankhede) [reply client/src/components/selfReviewDetails/SelfReviewDetails.js]: done

File: server/app/api/controllers/users.js (hunk 9 above)
Comment (RakshandaWankhede) [reply server/app/api/controllers/users.js]: done

**Comment by RakshandaWankhede** [comment]
same files are available in PR #31 , closing this PR

**Pull request closed without merging**